Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/corefinement.h

```diff
@@ -240,13 +255,92 @@ bool does_bound_a_volume(const TriangleMesh& tm)
+  * named parameters passed for output meshes should be done using `make_tuple()` as the types of
+  * named parameters are unspecified.
+  *
+  * If `tm1` and/or `tm2` are one of the output surface meshes, they will be updated to
+  * contain the output (in-place operation), in any other case, the corresponding result will
+  * be inserted into the mesh without clearing it first.
+  * \pre \link CGAL::Polygon_mesh_processing::does_self_intersect() `!CGAL::Polygon_mesh_processing::does_self_intersect(tm1)` \endlink
+  * \pre \link CGAL::Polygon_mesh_processing::does_self_intersect() `!CGAL::Polygon_mesh_processing::does_self_intersect(tm2)` \endlink
+  * \pre \link CGAL::Polygon_mesh_processing::does_bound_a_volume() `CGAL::Polygon_mesh_processing::does_bound_a_volume(tm1)` \endlink
+  * \pre \link CGAL::Polygon_mesh_processing::does_bound_a_volume() `CGAL::Polygon_mesh_processing::does_bound_a_volume(tm2)` \endlink
+  *
+  * @tparam TriangleMesh a model of `MutableFaceGraph`, `HalfedgeListGraph` and `FaceListGraph`
+  * @tparam NamedParameters1 a sequence of \ref pmp_namedparameters "Named Parameters"
+  * @tparam NamedParameters2 a sequence of \ref pmp_namedparameters "Named Parameters"
+  * @tparam NamedParametersOut0 a sequence of \ref pmp_namedparameters "Named Parameters" for computing the union of the volumes bounded by `tm1` and `tm2`
+  * @tparam NamedParametersOut1 a sequence of \ref pmp_namedparameters "Named Parameters" for computing the intersection of the volumes bounded by `tm1` and `tm2`
+  * @tparam NamedParametersOut2 a sequence of \ref pmp_namedparameters "Named Parameters" for computing the difference of the volumes bounded by `tm1` and `tm2`
+  * @tparam NamedParametersOut3 a sequence of \ref pmp_namedparameters "Named Parameters" for computing the difference of the volumes bounded by `tm2` and `tm1`
+  *
+  * @param tm1 first input triangulated surface mesh
+  * @param tm2 second input triangulated surface mesh
+  * @param output an array of output surface meshes
+  * @param np1 optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+  * @param np2 optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+  *
+  * \cgalNamedParamsBegin
+  *   \cgalParamBegin{vertex_point_map}
+  *     the property map with the points associated to the vertices of `tm1` (`tm2`).
+  *     If this parameter is omitted, an internal property map for
+  *     `CGAL::vertex_point_t` should be available in `TriangleMesh`
+  *   \cgalParamEnd
+  *   \cgalParamBegin{edge_is_constrained_map} a property map containing the
+  *     constrained-or-not status of each edge of `tm1` (`tm2`).
+  *   \cgalParamEnd
+  *   \cgalParamBegin{face_index_map} a property map containing the index of each face of `tm1` (`tm2`).
+  *     Note that if the property map is writable, the indices of the faces
+  *     of `tm1` and `tm2` will be set after the corefinement is done.
```

Comment:
you simply use the value

Reply:
surface_mesh, open_mesh yes. In general if you don't provide access to your indices and you don't set them in case of modification then indeed there is an issue (but that's the case with all algorithm modifying a mesh and assuming indices are present and valid).

Reply:
ok!